### ISSUE TITLE
Working towards an F4 1.0.3 release...

### ISSUE THREAD

**Issue by SlimerDude** [xored/f4#71]
Hi, I believe some of the current fixes, and ones previously made by Ivan, are quite significant - and it'd be good to see them released officially.

I was thinking that if the current pull requests are merged, then they'd be available in the [nightly build](http://download.xored.com/f4/nightly/) section. That would give people a chance to try them out.

Then assuming all is good, when Fantom 1.0.68 is released, a final tweak to update the `fantom-runtime` could be made. After that, would you consider a F4 1.0.3 release?

**Comment by haiodo** [comment]
I've pushed all PR's required to 1.0.3 and updated master to build against Eclipse Mars and DLTK 5.4. But I've see a problem with Fantom runtime startup, not sure why it happens. But it is not starts for first time. 

@SlimerDude could you please check local builded product against master on clean workspace, may be you could figure out why this is happen. 

Including my startup log:
[startup.txt](https://github.com/xored/f4/files/49622/startup.txt)

I think it could be because:
```
!ENTRY org.fantom 4 0 2015-12-02 16:46:22.051
!MESSAGE /Users/haiodo/Develop/git/f4/product/target/products/com.xored.f4.ide/macosx/cocoa/x86_64/F4.app/Contents/MacOS/qwe/.metadata/.plugins/org.fantom.sys/etc/sys/config.props (No such file or directory)
```
Some things are missing but required by fantom runtime on startup.

I will be happy to deploy 1.0.3 build then this issue will be resolved.

**Comment by SlimerDude** [comment]
Cool, I'll try to make time tomorrow morning.

**Comment by SlimerDude** [comment]
I would hold off releasing 1.0.3 until Fantom 1.0.68 is officially released. It would be good to have the final version in F4. I'm seeing Brian Frank next week so should get some resolution soon on when this may happen.

As for the startup error...

#### TLDR
Try applying this patch. I added a line to ensure the directory exists when copying sys files.
```
diff --git a/org.fantom/src/org/fantom/internal/core/FanHome.java b/org.fantom/src/org/fantom/internal/core/FanHome.java
index f4260b7..bb4b6a3 100644
--- a/org.fantom/src/org/fantom/internal/core/FanHome.java
+++ b/org.fantom/src/org/fantom/internal/core/FanHome.java
@@ -53,6 +53,7 @@ public class FanHome {
 		if (fileName.endsWith("/")) {
 			new File(location, fileName).mkdirs();
 		} else {
+			new File(location, fileName).getParentFile().mkdirs();
 			FileOutputStream os = new FileOutputStream(new File(location,
 					fileName));
 			copyStream(url.openStream(), os);
```

#### Explanation
What appears to be failing is `Sys.boot()` from `init()` in `org.fantom.sys/plugin-src/org/fantom/internal/sys/FantomInstall.java`

```java
  public void init() throws CoreException
  {
    Activator plugin = Activator.getDefault();
    FanHome home = new FanHome(plugin.getStateLocation().toFile());
    Bundle self = plugin.getBundle();
    home.init(self, "etc");
    home.init(self, "lib");
    System.setProperty("fan.home", home.getAbsolutePath());
    final Class<EquinoxEnv> envClass = EquinoxEnv.class;
    System.setProperty("FAN_ENV", "[java]" + envClass.getPackage().getName() + "::" + envClass.getSimpleName());
    Sys.boot();
  }
```

`Sys.boot()` expects files to be in the FAN_HOME dir. And that's what `home.init(...)` does, copies files into the dir. 

This all works fine in Indigo, so I guessed it's a Mars thing. And true enough, in Indigo `org.osgi.framework.Bundle.findEntries()` returns directory entries first, whereas in Mars they're returned at the end. 

Hence startup fails the first time because it couldn't copy files but DID create the directories. The 2nd time round, the dirs exist so the files got copied over.

**Comment by haiodo** [comment]
Thanks a lot, this worked.
Pushed to fantom-runtime and uploaded to http://updates.xored.com/fantom-runtime/

**Comment by haiodo** [comment]
Yes, sure, lets wait. I will try to find some one to check failed tests on Mars to be sure all is ok :-)

**Comment by SlimerDude** [comment]
Cool. Could you also kick-off a build in [Jenkins](https://cj.xored.com/job/f4/) so the fantom-runtime fix becomes part of the nightly build / download? Cheers.

**Comment by SlimerDude** [comment]
Just seem [build #24](http://download.xored.com/f4/nightly/) - thanks!

**Comment by SlimerDude** [comment]
See new commits / pull requests for the Fantom 1.0.68 / F4 1.0.3 release.

**Comment by haiodo** [comment]
Be able to make master compile against neon and pushed fixes related to build f4 from neon based f4 :-) Also updated DLKT to 5.5.  Most probable will upload 1.0.3 during this week.

**Comment by SlimerDude** [comment]
Given F4 v1.1 has been released, I vote to close this issue!
